### PR TITLE
Extract minifed CSS and JS from the index.php file

### PR DESCRIPTION
**Pull request by NicolasCARPi**
The minifed files are commited to the repo because one of the
requirement is to have it working out of the box.

The developers need to:

* install yarn
* run: yarn install
* run: grunt

This change should be reflected in the CONTRIBUTING page.

This needs to be done everytime the unminified version is changed (just the `grunt` step).

Also remove files in .gitignore that should be commited.
And split the CSS for admin page in its own file (they are combined afterwards): this improves separation of concern and readability for the dev.

With this change, the CSS and JS become accessible to humans, and also to the editor error checking:

2019-01-20-184417_948x645_scrot

2019-01-20-184435_1087x645_scrot

It is transparent to the final user, and brings much needed availability of the code to the developer.

It will also allow in the future to more easily add libraries and other cool things.

PS: what editor are you using? You should definitely use one with static error checking, it helps a lot finding small issues (like there was a bunch of missing semicolon in the JS, but because JS is very lenient it was fine, but it's better to add them). There was also a variable that was declared twice, I fixed that.

### CODE REVIEW

**Comment by robiso** [comment]
@NicolasCARPi, thanks for the awesome work. This is my first "conflict" I've ever seen. What's supposed to happen here. In layman terms, how can I fix this conflict of ours. :)
screenshot 2019-01-21 at 00 33 32

**Comment by NicolasCARPi** [comment]
A conflict happens when two changes of the same line don't agree. Then git shows you the two versions on top of each other separated by `>>>>>>`. And then you have to chose which version you want to keep. Because I did the changes prior to the `hints` branch merge, there was a difference between both versions of the comment of the `css` function.

I pulled your dev branch in my dev branch, then merged my dev branch into my extract branch, merged the conflict and pushed again.

You can now merge, there is no conflict anymore ;)

**Comment by robiso** [comment]
"Not so night owl" reporting, all good and thanks @NicolasCARPi :)
